### ISSUE TITLE
A web browser plugin and service to scan licenses while you browse

### ISSUE THREAD

**Issue by chinyeungli** [nexB/scancode-toolkit#2683]
## Short Description

When users see a license text in a webpage or license text/notice inside of an article, user can capture the text and save it in a file and then launch scancode detection to get the license information. However, wouldn't it be nice if user can either scan the whole webpage or selection of text directly from a brower plugin?

## Possible Labels

- new feature
- nice to have

## Select Category

-  [x] Enhancement
-  [ ] Add License/Copyright
-  [ ] Scan Feature
-  [ ] Packaging
-  [ ] Documentation
-  [ ] Expand Support
-  [ ] Other

**Comment by pombredanne** [comment]
That's an awesome feature.
And as an add-on I would also like to submit a PR or issue semi-automatically when there are detection issues that I want to correct.

Implementation-wise I can think of these ways:

1. a big fat browser plugin that contains all of scancode-toolkit including a Python runtime. The benefit would be that everything is self-contained and local.
2. same as above but one step further, as an Electron application, where the browsing takes place inside the app. It could be part of the scancode-workbench
3. a web-based mini app that accepts a text to scan and returns results (including some diff results) either as JSON or simple HTML; coupled with a lightweight browser plugin that interacts with this back-end web app.


I personally prefer 3. which  would also be usable without a browser plugin and could be also something that could be an API usable elsewhere too.

@alandtse FYI ... any update on the license changes for https://github.com/spdx/spdx-license-diff at https://github.com/spdx/spdx-license-diff/issues/7  ?

**Comment by alandtse** [comment]
> @alandtse FYI ... any update on the license changes for https://github.com/spdx/spdx-license-diff at [spdx/spdx-license-diff#7](https://github.com/spdx/spdx-license-diff/issues/7) ?

No, it's not really something I'm too motivated to spend time on to change from GPL-3.0 and no takers so far on doing it. That said, I didn't realize anyone really wanted to use any of my hacked together code. I could be convinced to be more explicit that the stuff I wrote could be Apache-2.0 while the `cc-by-sa.js` stuff is just that.

If you meant my other [issue about integrating better with scancode, including a submission feature](https://github.com/spdx/spdx-license-diff/issues/88), that also the same state of work. I haven't been too interested in working on the plugin but wouldn't be against accepting a PR by someone who wants to add a submit to scancode feature. It wouldn't be that much work if scancode code digest a [diffmatchpatch diff array](https://github.com/google/diff-match-patch/blob/master/javascript/diff_match_patch_uncompressed.js#L103) through it's api. All I'm doing is running the javascript version and then using it's [pretty html display function](https://github.com/google/diff-match-patch/blob/master/javascript/diff_match_patch_uncompressed.js#L1251).

EDIT: To clarify I'm not against changing the license, I'm just lazy. ;)

**Comment by pombredanne** [comment]
@alandtse ping ... still lazy?

**Comment by alandtse** [comment]
Yah still not actively developing. We're actually trying to find someone to help with the manifest deprecation for Chrome so that will take higher priority if we get any resources.

**Comment by pombredanne** [comment]
>  We're actually trying to find someone to help with the manifest deprecation for Chrome so that will take higher priority if we get any resources.

@alandtse Thanks for the reply!
there is still interest on our side, have you re-considered the license question?

**Comment by alandtse** [comment]
That was also a workload question of rewriting that code. I forget if we have the ability to relicense.
